File: 백준 문제 풀이 모음/class 4/1149.py
import sys

n = int(sys.stdin.readline())

result = [0,0,0]

for i in range(n) :
    tmp = list(map(int, sys.stdin.readline().split()))

    x1 = min(result[1] + tmp[0], result[2] + tmp[0])
    x2 = min(result[0] + tmp[1], result[2] + tmp[1])
    x3 = min(result[0] + tmp[2], result[1] + tmp[2])

    result[0] = x1
    result[1] = x2
    result[2] = x3


print(min(result))

File: 백준 문제 풀이 모음/kotlin/class 4/1149.kt


fun main() = with(System.`in`.bufferedReader()){
    val n = readLine().toInt()

    val matrix = MutableList<MutableList<Int>>(n) { MutableList<Int>(3) { 0 } }

    for (i in 0 until n) {
        val input = readLine().split(' ').toList()
        for (j in 0 until 3) {
            matrix[i][j] = input[j].toInt()
        }
    }

    for (i in 1 until n) {
        matrix[i][0] = Math.min(matrix[i][0] + matrix[i-1][1], matrix[i][0] + matrix[i-1][2])
        matrix[i][1] = Math.min(matrix[i][1] + matrix[i-1][0], matrix[i][1] + matrix[i-1][2])
        matrix[i][2] = Math.min(matrix[i][2] + matrix[i-1][0], matrix[i][2] + matrix[i-1][1])
    }

    println(matrix[n-1].min())


}
